towerlib/towerlib.py

```diff
@@ -1997,24 +2034,707 @@ def schedules(self):
+            for project in matching_projects:
+                self.update_project_by_id(project.id)
+            self._logger.debug("Updated {} projects with given url '{}'".format(len(matching_projects), scm_url))
+        else:
+            self._logger.debug("No project was updated. Reason: no tower project found with "
+                               "the given url '{}'".format(scm_url))
+
+    def update_project_by_branch_name(self, scm_url, branch_name):
+        """Update an ansible tower project or list of projects based on their branch name.
+
+        A scm_branch can only be identified correctly with a corresponding scm_url.
+
+        Args:
+            scm_url: the URL of the relevant repository configured in the project.
+            branch_name: the name of the branch, which is selected as scm_branch parameter of the project.
+
+        Returns:
+            list: List of project update ids on success, otherwise empty list.
+
+        """
+        projects = self.get_all_projects()
+        project_update_id_list = []
+        matching_projects = [project for project in projects if
+                             project.scm_url == scm_url and project.scm_branch == branch_name]
+        if len(matching_projects) > 0:
+            for project in matching_projects:
+                self._logger.info("Updating the project: {}".format(project.name))
+                project_update = self.update_project_by_id(project.id)
+                project_update_id_list.append(project_update['id'])
+            self._logger.info("Updated {} project(s) with the url '{}' and branch name '{}'".
+                              format(len(matching_projects), scm_url, branch_name))
+            return project_update_id_list
+        else:
+            self._logger.warning("No project was updated. Reason: no tower project found with the given "
+                                 "git url '{}' and branch '{}'".format(scm_url, branch_name))
+            return []
+
+    def get_all_job_templates(self):
+        """Get all the job templates of the ansible tower.
+
+        Returns:
+            list: the list of all job templates as EntityManager object.
+
+        """
+        job_templates = [item for item in self.job_templates]
+        return job_templates
+
+    def get_job_templates_by_project(self, project_name):
+        """Get all the job templates for a given project name.
+
+        Args:
+            project_name: the complete api url of the project.
+
+        Returns:
+            list: a list of all the job templates object for the given project name.
+
+        """
+        required_job_templates = []
+        job_templates = self.get_all_job_templates()
+        for job_template in job_templates:
+            if job_template.project.name == project_name:
+                required_job_templates.append(job_template)
+        return required_job_templates
+
+    def change_job_template_data(self, job_template_data):
```

Comment:
This should be refactored with the above changes in mind, as a method of the job_template object.

Reply:
Implemented with b0db88a5e65f1ce0a8c96b5f5a35af546d5d2658